.github/workflows/healthCheckForNewNetworkDeployment.yml

```diff
@@ -0,0 +1,101 @@
+name: Health Check for New Network Deployment
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - 'config/networks.json'
+
+jobs:
+  check-new-network-health:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 
+
+      - name: Check if config/networks.json was changed in this branch
+        id: check-file-change
+        run: |
+          if git diff --name-only origin/main...HEAD | grep -q "config/networks.json"; then
+            echo "config/networks.json has been modified in this branch"
+            echo "CONTINUE=true" >> $GITHUB_ENV
+          else
+            echo "No changes in config/networks.json detected in this branch"
+            echo "CONTINUE=false" >> $GITHUB_ENV
+          fi
+
+      - name: Detect Newly Added Networks
+        if: env.CONTINUE == 'true'
+        id: detect-changes
+        run: |
+          echo "Comparing config/networks.json with the previous commit..."
+          git fetch origin main --depth=1 || echo "No previous commit found."
+
+          if git show origin/main:config/networks.json > /dev/null 2>&1; then
+            OLD_NETWORKS=$(git show origin/main:config/networks.json | jq 'keys')
+          else
+            echo "No previous networks.json found, assuming all networks are new."
+            OLD_NETWORKS="[]"
+          fi
+
+          NEW_NETWORKS=$(jq 'keys' config/networks.json)
+
+          echo "Old Networks: $OLD_NETWORKS"
```

Comment:
maybe only log the added networks? Or do you see a benefit of logging first old and new and then added networks?